src/components/i18n/Param.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+import * as React from 'react';
+
+type Props = {
+    /** The value of this parameter */
+    description: string,
+
+    /** A description of this parameter to help the translators understand what it is */
+    value: any,
+};
+
+/**
+ * @class A placeholder for a replacement parameter in the body of a FormattedCompMessage
+ * component.
+ *
+ * This component renders into the value of the named parameter to the FormattedCompMessage
+ * component. Typically, this component is self-closing.
+ *
+ * @example
+ * <pre>
+ *   <FormattedCompMessage id="x" description="y">
+ *     The file <Param value={filelist[i].path} description="Name of the file that was deleted."/> has been deleted.
+ *   </FormattedCompMessage>
+ * </pre>
+ */
+class Param extends React.Component<Props> {
+    getValue() {
+        switch (typeof this.props.value) {
+            default:
+            case 'undefined':
+                return '';
+
+            case 'boolean':
+            case 'number':
+                return String(this.props.value);
+
+            case 'function':
+                return this.props.value();
+
+            case 'string':
```

Comment:
We should also make the strings into constants and share it between the files. The other file also uses them.